extras/falcor-perf-extract-results.py

```diff
@@ -0,0 +1,170 @@
+    pkc = re.findall(r"(?<=Time for program kernel creation )(\((glslang|slang)\): ([0-9]*\.[0-9][0-9][0-9]))", data)   # Program kernel creation
+    fe = re.findall(r"(?<=Time for frontend execution:)([0-9]*\.[0-9][0-9][0-9])", data)                                # Frontend execution
+    sg = re.findall(r"(?<=Time for spirv generation by )((glslang|slang): ([0-9]*\.[0-9][0-9][0-9]))", data)            # Spirv generation
+    sc = re.findall(r"(?<=Time for compiling spirv generated by )((glslang|slang): ([0-9]*\.[0-9][0-9][0-9]))", data)   # Spirv compilation
+
+    rtn_json[results_id]["program-version-creation"] = {pvc[0][1]: float(pvc[0][2]), pvc[1][1]: float(pvc[1][2])}
+    rtn_json[results_id]["program-kernel-creation"] = {pkc[0][1]: float(pkc[0][2]), pkc[1][1]: float(pkc[1][2])}
+    rtn_json[results_id]["frontend-execution"] = {"glslang": float(fe[0]), "slang": float(fe[1])}
+    rtn_json[results_id]["spirv-generation"] = {sg[0][1]: float(sg[0][2]), sg[1][1]: float(sg[1][2])}
+    rtn_json[results_id]["spirv-compilation"] = {sc[0][1]: float(sc[0][2]), sc[1][1]: float(sc[1][2])}
+    rtn_json[results_id]["program-version-creation"] = {pvc[0][1]: float(pvc[0][2]), pvc[1][1]: float(pvc[1][2])}
+    rtn_json[results_id]["program-kernel-creation"] = {pkc[0][1]: float(pkc[0][2]), pkc[1][1]: float(pkc[1][2])}
+    rtn_json[results_id]["frontend-execution"] = {"glslang": float(fe[0]), "slang": float(fe[1])}
+    rtn_json[results_id]["spirv-generation"] = {sg[0][1]: float(sg[0][2]), sg[1][1]: float(sg[1][2])}
+    rtn_json[results_id]["spirv-compilation"] = {sc[0][1]: float(sc[0][2]), sc[1][1]: float(sc[1][2])}
+    program_version_creation_glslang.append(float(rtn_json[results_id]["program-version-creation"]["glslang"]))
+    program_version_creation_slang.append(float(rtn_json[results_id]["program-version-creation"]["slang"]))
+    program_kernel_creation_glslang.append(float(rtn_json[results_id]["program-kernel-creation"]["glslang"]))
+    program_kernel_creation_slang.append(float(rtn_json[results_id]["program-kernel-creation"]["slang"]))
+    frontend_execution_glslang.append(float(rtn_json[results_id]["frontend-execution"]["glslang"]))
+    frontend_execution_slang.append(float(rtn_json[results_id]["frontend-execution"]["slang"]))
+    spirv_generation_glslang.append(float(rtn_json[results_id]["spirv-generation"]["glslang"]))
+    spirv_generation_slang.append(float(rtn_json[results_id]["spirv-generation"]["slang"]))
+    spirv_compilation_glslang.append(float(rtn_json[results_id]["spirv-compilation"]["glslang"]))
+    spirv_compilation_slang.append(float(rtn_json[results_id]["spirv-compilation"]["slang"]))
+
+# In order to have charts in the spreadsheet that automatically update to the most recent
+# 30 test runs, we need to check if we already have 30 runs worth of data. Deleting and
+# re-inserting rows breaks the charts. So, instead we need to update the cells one by one.
+if len(recent_records) > 29:
```

Comment:
Instead of deleting data, can we just preserve all the data and create charts from google sheets using only the last 30 rows? That way we can view last 30 days, last year etc. if we want.

Reply:
I couldn't find a way for google sheets to automatically use the last 30 rows. However, no data is lost. I have one sheet for the raw data from every run, one sheet that has all the averages, then there is the sheet for the last 30 runs. Only the sheet for the last 30 runs ever has data removed, but that data is still available on the other sheets.

Reply:
I see, that sounds good!